File: studyworld-frontend/src/services/enrollmentService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/enrollments';

const enrollmentService = {
    // Fetch all enrollments
    getAllEnrollments: async () => {
        const response = await axios.get(API_URL);
        return response.data;
    },

    // Enroll a user in a course
    enrollUserInCourse: async (userId, courseId) => {
        const response = await axios.post(`${API_URL}/user/${userId}/course/${courseId}`);
        return response.data;
    },

    // Get enrollments by user ID
    getEnrollmentsByUserId: async (userId) => {
        const response = await axios.get(`${API_URL}/user/${userId}`);
        return response.data;
    },

    // Get enrollments by course ID
    getEnrollmentsByCourseId: async (courseId) => {
        const response = await axios.get(`${API_URL}/course/${courseId}`);
        return response.data;
    },

    // Delete an enrollment
    deleteEnrollment: async (enrollmentId) => {
        await axios.delete(`${API_URL}/${enrollmentId}`);
    }
};

export default enrollmentService;

File: studyworld/src/main/java/com/tutorials/studyworld/service/EnrollmentService.java
package com.tutorials.studyworld.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import com.tutorials.studyworld.model.Enrollment;
import com.tutorials.studyworld.model.User;
import com.tutorials.studyworld.model.Course;
import com.tutorials.studyworld.repository.EnrollmentRepository;
import com.tutorials.studyworld.repository.UserRepository;
import com.tutorials.studyworld.repository.CourseRepository;

import java.util.List;

@Service
public class EnrollmentService {

    @Autowired
    private EnrollmentRepository enrollmentRepository;

    @Autowired
    private UserRepository userRepository;

    @Autowired
    private CourseRepository courseRepository;

    public List<Enrollment> getAllEnrollments() {
        return enrollmentRepository.findAll();
    }

    public Enrollment getEnrollmentById(Long id) {
        return enrollmentRepository.findById(id).orElse(null);
    }

    public List<Enrollment> getEnrollmentsByUserId(Long userId) {
        return enrollmentRepository.findByUserId(userId);
    }

    public List<Enrollment> getEnrollmentsByCourseId(Long courseId) {
        return enrollmentRepository.findByCourseId(courseId);
    }

    public Enrollment enrollUserInCourse(Long userId, Long courseId) {
        User user = userRepository.findById(userId).orElse(null);
        Course course = courseRepository.findById(courseId).orElse(null);
        if (user != null && course != null) {
            Enrollment enrollment = new Enrollment();
            enrollment.setUser(user);
            enrollment.setCourse(course);
            return enrollmentRepository.save(enrollment);
        }
        return null;
    }

    public void unenrollUserFromCourse(Long id) {
        enrollmentRepository.deleteById(id);
    }
}
